crates/vizia_core/src/modifiers/actions.rs

```diff
@@ -1,634 +1,551 @@
+    pub(crate) on_hover: Option<Arc<dyn Fn(&mut EventContext) + Send + Sync>>,
+    pub(crate) on_hover_out: Option<Arc<dyn Fn(&mut EventContext) + Send + Sync>>,
+    pub(crate) on_over: Option<Arc<dyn Fn(&mut EventContext) + Send + Sync>>,
+    pub(crate) on_over_out: Option<Arc<dyn Fn(&mut EventContext) + Send + Sync>>,
+    pub(crate) on_mouse_move: Option<Arc<dyn Fn(&mut EventContext, f32, f32) + Send + Sync>>,
+    pub(crate) on_mouse_down: Option<Arc<dyn Fn(&mut EventContext, MouseButton) + Send + Sync>>,
+    pub(crate) on_mouse_up: Option<Arc<dyn Fn(&mut EventContext, MouseButton) + Send + Sync>>,
+    pub(crate) on_focus_in: Option<Arc<dyn Fn(&mut EventContext) + Send + Sync>>,
+    pub(crate) on_focus_out: Option<Arc<dyn Fn(&mut EventContext) + Send + Sync>>,
+    pub(crate) on_geo_changed:
+        Option<Arc<dyn Fn(&mut EventContext, GeometryChanged) + Send + Sync>>,
 }
 
-impl<V: View> Press<V> {
-    pub fn new<'a, F>(handle: Handle<'a, V>, action: F) -> Handle<'a, Press<V>>
-    where
-        F: 'static + Fn(&mut EventContext),
-    {
-        if let Some(mut view) = handle.cx.views.remove(&handle.entity) {
-            if view.downcast_ref::<V>().is_some() {
-                let item = Self { view, action: Some(Box::new(action)), p: Default::default() };
-
-                handle.cx.views.insert(handle.entity, Box::new(item));
-            } else {
-                if let Some(press) = view.downcast_mut::<Press<V>>() {
-                    press.action = Some(Box::new(action));
-                }
-                handle.cx.views.insert(handle.entity, view);
-            }
+impl ActionsModel {
+    pub(crate) fn new() -> Self {
+        Self {
+            on_press: None,
+            on_press_down: None,
+            on_hover: None,
+            on_hover_out: None,
+            on_over: None,
+            on_over_out: None,
+            on_mouse_move: None,
+            on_mouse_down: None,
+            on_mouse_up: None,
+            on_focus_in: None,
+            on_focus_out: None,
+            on_geo_changed: None,
         }
-
-        Handle { entity: handle.entity, p: Default::default(), cx: handle.cx }
     }
 }
 
-impl<V: View> View for Press<V> {
-    fn element(&self) -> Option<&'static str> {
-        self.view.element()
-    }
-
+impl Model for ActionsModel {
     fn event(&mut self, cx: &mut EventContext, event: &mut Event) {
-        self.view.event(cx, event);
-
-        event.map(|window_event, _| match window_event {
-            WindowEvent::Press { mouse } => {
-                let over = if *mouse { cx.hovered() } else { cx.focused() };
-                if cx.current() != over && !over.is_descendant_of(cx.tree, cx.current()) {
-                    return;
-                }
-                if let Some(action) = &self.action {
-                    (action)(cx);
-                }
+        event.map(|actions_event, _| match actions_event {
```

Comment:
this seems like the perfect opportunity for `event.take()`. Would that also let us get rid of the `Arc`?

Reply:
Good idea! I think I originally wrote this part before `event.take()` was merged and then I completely forgot it existed. I'll give this a go and see if we can renove those `Arc`s.

Reply:
I made the suggested change to use `event.take()` and I was able to remove the clones but I still need to `Box` the closures unless I'm missing something? In any case it did remove the need for `Arc`.